rskj-core/src/test/java/co/rsk/rpc/modules/personal/PersonalModuleTest.java

```diff
@@ -0,0 +1,69 @@
+package co.rsk.rpc.modules.personal;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.ethereum.core.Transaction;
+import org.ethereum.core.TransactionPoolAddResult;
+import org.ethereum.datasource.HashMapDB;
+import org.ethereum.facade.Ethereum;
+import org.ethereum.rpc.TypeConverter;
+import org.ethereum.rpc.Web3;
+import org.junit.Test;
+
+import co.rsk.config.TestSystemProperties;
+import co.rsk.core.RskAddress;
+import co.rsk.core.Wallet;
+
+public class PersonalModuleTest {
+
+	private static final String PASS_FRASE = "passfrase";
+	
+    @Test
+    public void sendTransactionWithGasLimitTest() throws Exception {
+    	
+    	TestSystemProperties props = new TestSystemProperties();
+    	
+    	Wallet wallet = new Wallet(new HashMapDB());
+    	RskAddress sender = wallet.addAccount(PASS_FRASE);
+    	RskAddress receiver = wallet.addAccount();
+    	
+    	// Simulation of the args handled in the sendTransaction call
+    	Web3.CallArguments args = new Web3.CallArguments();
+    	args.from = sender.toJsonString();
+    	args.to = receiver.toJsonString();
+    	args.gasLimit = "0x76c0";
+    	args.gasPrice = "0x9184e72a000";
+    	args.value = "0x186A0";
+    	args.nonce = "0x01";
+
+    	// Transaction that is expected to be constructed WITH the gasLimit
+        Transaction tx = Transaction
+                .builder()
+                .nonce(TypeConverter.stringNumberAsBigInt(args.nonce))
+                .gasPrice(TypeConverter.stringNumberAsBigInt(args.gasPrice))
+                .gasLimit(TypeConverter.stringNumberAsBigInt(args.gasLimit))
+                .destination(TypeConverter.stringHexToByteArray(args.to))
+                .chainId(props.getNetworkConstants().getChainId())
+                .value(TypeConverter.stringNumberAsBigInt(args.value))
+                .build();
+        tx.sign(wallet.getAccount(sender, PASS_FRASE).getEcKey().getPrivKeyBytes());
+
+        // Hash of the expected transaction
+        String txExpectedResult = tx.getHash().toJsonString();
+    	
+    	TransactionPoolAddResult transactionPoolAddResult = mock(TransactionPoolAddResult.class);
```

Comment:
could you please align code in this and some other files? Seems you've used a mix of tab and space symbols ) If Im not mistaken, the indent is 4 spaces in the code.

Reply:
It is in the exact same way that it was in the EthModuleTransactionBase

Reply:
Same here, lets try to avoid duplicated code or at least open it as tech debt. We're executing the same transaction in the same way, just by different inputs

Something like this can be achieved
```java 
protected void extracted(Wallet wallet, 
                             Constants constants, 
                             TransactionGateway transactionGateway, 
                             TransactionPoolAddResult transactionPoolAddResult, 
                             TransactionPool transactionPool, 
                             EthModuleTransactionBase ethModuleTransaction, 
                             RskAddress sender, 
                             RskAddress receiver, 
                             Function<Web3.CallArguments, String> sendTransactionFunc, 
                             byte[] privKeyBytes) {

        // Simulation of the args handled in the sendTransaction call
        Web3.CallArguments args = new Web3.CallArguments();
        args.from = sender.toJsonString();
        args.to = receiver.toJsonString();
        args.gasLimit = "0x76c0";
        args.gasPrice = "0x9184e72a000";
        args.value = "0x186A0";
        args.nonce = "0x01";

        Transaction tx = Transaction
                .builder()
                .nonce(TypeConverter.stringNumberAsBigInt(args.nonce))
                .gasPrice(TypeConverter.stringNumberAsBigInt(args.gasPrice))
                .gasLimit(TypeConverter.stringNumberAsBigInt(args.gasLimit))
                .destination(TypeConverter.stringHexToByteArray(args.to))
                .chainId(constants.getChainId())
                .value(TypeConverter.stringNumberAsBigInt(args.value))
                .build();
        tx.sign(privKeyBytes);

        // Hash of the expected transaction
        String txExpectedResult = tx.getHash().toJsonString();
        
        // Hash of the actual transaction builded inside the sendTransaction
        String txResult = sendTransactionFunc.apply(args);

        assertEquals(txExpectedResult, txResult);
    }
```

Reply:
Nice, i will open the issue!